File: Cantidad de dígitos de un número/Java/Cuenta_Digitos_v4.java
package cuenta_digitos_v4;

import static java.lang.Math.abs;
import java.util.Scanner;

public class Cuenta_Digitos_v4 {

    public static void main(String[] args) {
        int numero;
        Scanner reader = new Scanner(System.in);
        System.out.printf("Ingrese un número: ");
        numero = reader.nextInt();

        numero = abs(numero);
        System.out.printf("El número tiene %d dígito(s)%n", numero == 0 ? 1 : cuenta_digitos(numero));
    }

    private static int cuenta_digitos(int numero) {
        if (numero == 0) {
            return 0;
        }
        return 1 + cuenta_digitos(numero / 10);
    }

}

File: Cantidad de dígitos de un número/Python/cuenta_digitos_v4.py
#Creado por Andrés Melgar
#para iterando++ http://iterando.online/
#entrada: http://iterando.online/ejercicios-resueltos/cantidad-de-digitos-de-un-numero-en-python/
import math

def cuenta_digitos(numero, base):
    if numero == 0:
        return 1
    return int(math.log(numero)+math.log(base)) + 1

numero=int(input("Ingrese un número: "))
base=int(input("Ingrese una base: "))
numero=abs(numero)
if base>1:
    print("El número tiene {:d} dígito(s) en la base {:d}".format(cuenta_digitos(numero, base), base))
else:
    print("La base debe ser mayor que 1")